src/Network/Network/LoadBalancer/NewAzLoadBalancerBackendAddressConfig.cs

```diff
@@ -0,0 +1,100 @@
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using CNM = Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using System.Net;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "LoadBalancerBackendAddressConfig"), OutputType(typeof(PSLoadBalancerBackendAddress))]
+    public partial class NewAzureLoadBalancerBackendAddress : NetworkBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The IPAddress to add to the backend pool",
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public string IpAddress { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the Backend Address config",
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The network interface IP config",
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public PSNetworkInterfaceIPConfiguration NetworkInterfaceIPConfiguration { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The virtual network associated with Backend Address config",
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public PSVirtualNetwork VirtualNetwork { get; set; }
+
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (!ValidateIpAddress(this.IpAddress))
+            {
+                throw new PSArgumentException($"Invalid IPAddress : {Properties.Resources.InvalidIPAddress}");
+            }
+
+            var loadBalancerBackendAddress = new PSLoadBalancerBackendAddress();
+            loadBalancerBackendAddress.BackendIpConfigurations = this.NetworkInterfaceIPConfiguration;
+            loadBalancerBackendAddress.VirtualNetwork = this.VirtualNetwork;
+            loadBalancerBackendAddress.IpAddress = this.IpAddress;
+            loadBalancerBackendAddress.Name = this.Name;
+
+            WriteObject(loadBalancerBackendAddress);
+        }
+
+        public bool ValidateIpAddress(string ipAddress)
+        {
+            IPAddress result = null;
```

Comment:
this comes after if?

Reply:
Yeah, that's fine. If its not a empty string, it validates if its a proper IpAddress.